apps/emqx/src/emqx_channel.erl

```diff
@@ -2088,6 +2105,23 @@ is_clientid_namespaced(ClientId, Tns) when is_binary(Tns) andalso Tns =/= <<>> -
 is_clientid_namespaced(_ClientId, _Tns) ->
     false.
 
+%%--------------------------------------------------------------------
+%% Adjust limiter
+
+adjust_limiter(_ConnPkt, #channel{clientinfo = ClientInfo} = Channel0) ->
+    #{zone := Zone, listener := ListenerId} = ClientInfo,
+    maybe
+        Tns = get_tenant_namespace(ClientInfo),
+        true ?= Tns /= undefined,
```

Comment:
this assumes the limiter adjust only happens for multi-tenancy.
i think it's more logical that here in emqx_channel, it just evaluates the callback without checks.
i.e. it's the `emqx_mt` callback's job to decide if adjust should happen or not based on `Tns =/= undefined`.